### PR TITLE
[CH] A simple job scheduler for merge tree cache sync load

### PR DESCRIPTION
**Pull request by liuneng1994**
## What changes were proposed in this pull request?

When the cache is loaded synchronously, the time consumed may be greater than the timeout of the spark rpc. A new asynchronous task mechanism is introduced to implement cache synchronous loading through polling, and a unified exception handling is added.

## How was this patch tested?

unit tests

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Review by loneylee — APPROVED** [review]

File: backends-clickhouse/src/main/scala/org/apache/spark/sql/execution/commands/GlutenCHCacheDataCommand.scala [hunk 1]
```diff
@@ -100,14 +102,14 @@ case class GlutenCHCacheDataCommand(
       selectedColuman.get
         .filter(allColumns.contains(_))
         .map(ConverterUtils.normalizeColName)
-        .toSeq
```
Comment (zzcclp) [inline backends-clickhouse/src/main/scala/org/apache/spark/sql/execution/commands/GlutenCHCacheDataCommand.scala:103]: don't remove this, will failed with the Scala 2.13

File: backends-clickhouse/src/main/scala/org/apache/spark/sql/execution/commands/GlutenCHCacheDataCommand.scala [hunk 2]
```diff
@@ -201,87 +203,112 @@ case class GlutenCHCacheDataCommand(
...
+  def waitAllJobFinish(jobs: ArrayBuffer[(String, CacheJobInfo)]): (Boolean, String) = {
+    val res = collectJobTriggerResult(jobs)
+    var status = res._1
+    val messages = res._2
+    jobs.foreach(
+      job => {
+        if (status) {
+          var complete = false
+          while (!complete) {
+            Thread.sleep(5000)
```
Comment (zzcclp) [inline backends-clickhouse/src/main/scala/org/apache/spark/sql/execution/commands/GlutenCHCacheDataCommand.scala:259]: 5000 ? consider to reduce this value to 1000 or less

Comment (liuneng1994) [reply backends-clickhouse/src/main/scala/org/apache/spark/sql/execution/commands/GlutenCHCacheDataCommand.scala:259]: 预期异步加载时间是分钟级到小时级，频率不宜过高

**Comment by liuneng1994** [comment]
Run Gluten Clickhouse CI

**Review by zzcclp — APPROVED** [review]
LGTM

**Review by zzcclp — APPROVED** [review]
LGTM

**Review by zzcclp — APPROVED** [review]
LGTM

**Pull request merged**